src/app/modules/submission/email-submission/email-submission.service.ts

```diff
@@ -0,0 +1,46 @@
+import {
+  isProcessedCheckboxResponse,
+  isProcessedTableResponse,
+} from '../../../utils/field-validation/field-validation.guards'
+import { ProcessedFieldResponse } from '../submission.types'
+
+import { EmailData, EmailDataForOneField } from './email-submission.types'
+import {
+  getAnswerForCheckbox,
+  getAnswerRowsForTable,
+  getFormattedResponse,
+} from './email-submission.util'
+
+export const createEmailData = (
+  parsedResponses: ProcessedFieldResponse[],
+  hashedFields: Set<string>,
+): EmailData => {
+  const emailData: EmailData = {
+    autoReplyData: [],
+    jsonData: [],
+    formData: [],
+  }
+  parsedResponses.forEach((response) => {
+    let formattedResponses: EmailDataForOneField[]
+    if (isProcessedTableResponse(response)) {
+      formattedResponses = getAnswerRowsForTable(response).map((row) =>
+        getFormattedResponse(row, hashedFields),
+      )
+    } else if (isProcessedCheckboxResponse(response)) {
+      const checkbox = getAnswerForCheckbox(response)
+      formattedResponses = [getFormattedResponse(checkbox, hashedFields)]
+    } else {
+      formattedResponses = [getFormattedResponse(response, hashedFields)]
+    }
+    formattedResponses.forEach((formattedResponse) => {
```

Comment:
Can we use `.reduce` instead? Don't know if it's overkill or makes it harder to understand LOL...

```ts
const emailData = formattedResponses
  .reduce((emailData, current) => {
    if (current.autoReplyData) emailData.autoReplyData.push(current.autoReplyData)
    if (current.jsonData) emailData.jsonData.push(current.jsonData)
    emailData.formData.push(current.formData)

    return emailData
  }, originalEmailData)
```


this can even be chained with the previous map step

Reply:
I've changed it to reduce but I personally think it's more confusing than before, plus requires 3 iterations through the data instead of 1. lmk what you think?